spec/acceptance/suites/two_domains/00_rsyslog_forward_spec.rb

```diff
@@ -0,0 +1,183 @@
+require 'spec_helper_acceptance'
+
+test_name 'simp_rsyslog profile'
+
+describe 'simp_rsyslog' do
+  before(:context) do
+    hosts.each do |host|
+      interfaces = fact_on(host, 'interfaces').strip.split(',')
+      interfaces.delete_if do |x|
+        x =~ /^lo/
+      end
+
+      interfaces.each do |iface|
+        if fact_on(host, "ipaddress_#{iface}").strip.empty?
+          on(host, "ifup #{iface}", :accept_all_exit_codes => true)
+        end
+      end
+    end
+  end
+
+  let(:manifest) { <<-EOS
+include 'simp_rsyslog'
+EOS
+  }
+  rsyslog_server1 = hosts_with_role(hosts,'rsyslog_server1').first
+  rsyslog_server2 = hosts_with_role(hosts,'rsyslog_server2').first
+  let(:server1_fqdn){fact_on(rsyslog_server1, 'fqdn')}
+  let(:server2_fqdn){fact_on(rsyslog_server2, 'fqdn')}
+  let(:server_hieradata) {
+      <<-EOS
+---
+simp_options::syslog::log_servers:
+  - '#{server1_fqdn}'
+  - '#{server2_fqdn}'
+
+simp_rsyslog::is_server: true
+simp_rsyslog::forward_logs: false
+rsyslog::tcp_server: true
+rsyslog::tls_tcp_server: true
+rsyslog::pki: true
+rsyslog::app_pki_external_source: '/etc/pki/simp-testing/pki'
+# Need to let log servers accept from different domains
+rsyslog::config::tls_input_tcp_server_stream_driver_permitted_peers:
+  - "*.wayout.org"
+  - "*.my.domain"
+EOS
+  }
+  # Set up the servers first so they can both receive from the clients
+  hosts_with_role(hosts, 'rsyslog_server').each do |server|
+
+    it "should configure server #{server} without errors" do
+      set_hieradata_on(server, server_hieradata)
+      apply_manifest_on(server, manifest, :catch_failures => true)
+    end
+
+    it "should configure #{server} idempotently" do
+      apply_manifest_on(server, manifest, :catch_changes => true)
+    end
+  end
+
+  hosts_with_role(hosts, 'rsyslog_server').each do |server|
+    hosts_with_role(hosts, 'rsyslog_client').each do |client|
+      context "#{client} logging to the remote syslog server #{server} with default forwarding" do
+        let(:client_fqdn){ fact_on( client, 'fqdn' ) }
+        let(:client_hieradata) {
+          <<-EOS
+---
+simp_options::syslog::log_servers:
+  - '#{server1_fqdn}'
+  - '#{server2_fqdn}'
+rsyslog::app_pki_external_source: '/etc/pki/simp-testing/pki'
+simp_rsyslog::forward_logs: true
+rsyslog::pki: true
+rsyslog::enable_tls_logging: true
+         EOS
+        }
+
+        let(:client_log_dir) { "/var/log/hosts/#{client_fqdn}" }
+
+        it "should configure client #{client} without errors" do
+          set_hieradata_on(client, client_hieradata)
+          apply_manifest_on(client, manifest, :catch_failures => true)
+        end
+
+        it "should configure #{client} idempotently" do
+          apply_manifest_on(client, manifest, :catch_changes => true)
+        end
+
```

Comment:
This test doesn't need to test all the forwarding rules, just one forwarding rule.  Another test already handles the in-depth rule forwarding.